archetypes/catalog/pom.xml

```diff
@@ -67,4 +67,95 @@
             <optional>true</optional>
         </dependency>
     </dependencies>
+
+    <properties>
+        <!-- support command-line overrides -->
+        <build.tools.version>${version.plugin.helidon-build-tools}</build.tools.version>
+    </properties>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.helidon.build-tools</groupId>
+                <artifactId>helidon-stager-maven-plugin</artifactId>
+                <version>${version.plugin.helidon-build-tools}</version>
+                <executions>
+                    <execution>
+                        <phase>install</phase>
+                        <goals>
+                            <goal>stage</goal>
+                        </goals>
+                        <configuration>
+                            <directories>
+                                <directory target="${project.build.directory}">
+                                    <archives>
+                                        <archive target="cli-data/${project.version}/cli-data.zip">
+                                            <unpack-artifacts>
+                                                <unpack-artifact groupId="io.helidon.archetypes"
+                                                                 artifactId="helidon-archetype-catalog"
+                                                                 version="${project.version}"
+                                                                 type="zip"
+                                                                 target="."/>
+                                            </unpack-artifacts>
+                                            <templates>
+                                                <template source="cli-metadata.properties.mustache"
+                                                          target="metadata.properties">
+                                                    <variables>
+                                                        <variable name="buildtoolsVersion"
+                                                                  value="${build.tools.version}"/>
```

Comment:
Can you use the same name of in helidon-site ? I.e. `cli.maven.plugin.version`
I know we have `build.tools.version` in `metadata.properties`, but IMO this is too confusing to expose as property.
E.g. The line below has `${version.plugin.helidon-build-tools}`

I would like to use `cli.maven.plugin.version` onwards and look at renaming the property inside `metadata.properties` (even if that means having code that support both properties to maintain backward compatibility).

Also this property needs to have a default value since this stager plugin execution is added by default.
You should define the property with a default value in the top level pom:

```
<cli.maven.plugin.version>${version.plugin.helidon-build-tools}</cli.maven.plugin.version>
```

Reply:
`cli.maven.plugin.version` bothers me first because it doesn't following the normal naming convention `version.plugin.*` and second because there is no actual corresponding `cli` artifact; I didn't change it to better match the metadata property name. (Also, I still think we should be using this value when we generate from the archetypes, and then this name would be accurate.)

But I will change it just to make you happy ;-).

Reply:
![image](https://user-images.githubusercontent.com/4099370/86648715-f46c6c00-bf95-11ea-82bf-a26bfb7c355d.png)

Reply:
My main issue with `build-tools.version` is that it's too generic. It does not follow the convention so you could use `version.plugin.helidon.cli`, or since as you point out there is no such maven artifact, you can use the existing property used to drive the helidon maven plugin: `version.plugin.helidon-build-tools`.

Reply:
The changes are already in to switch to `cli.maven.plugin.version`.

Reply:
Right, but as you noted there it does not follow the current convention and does not really point to a maven artifact.

Reply:
We agreed on `cli.build.tools.version` and that change has been pushed.